File: JavaScript/context.js
class Context {
  env = {};
  parent = null;
  constructor(env, parent) {
    this.env = env;
    this.parent = parent;
  }
  get(key) {
    if (key in this.env) {
      return this.env[key];
    }
    if (this.parent) return this.parent.get(key);
    return null;
  }
  add(key, value) {
    if (!env) env = {};
    env[key] = value;
  }
}

export default Context;

File: python/context.py
class Context:
    def __init__(self, env, parent) -> None:
        self.env = {}
        self.parent = None
        if env:
            self.env = env
        self.parent = parent

    def get(self, key):
        if key in self.env:
            return self.env[key]
        if self.parent:
            return self.parent.get(key)
        return None

    def set(self, key, value) -> None:
        self.env[key] = value
